Tests/LibCrypto/TestHash.cpp

```diff
@@ -166,6 +175,15 @@ TEST_CASE(test_SHA256_hash_empty_string)
     EXPECT(memcmp(result, digest.data, Crypto::Hash::SHA256::digest_size()) == 0);
 }
 
+TEST_CASE(test_SHA256_hash_split_into_blocks)
+{
+    u8 result[] {
+        0x1b, 0x90, 0x74, 0xf9, 0x9b, 0xf3, 0x56, 0x20, 0xd3, 0xf6, 0xbf, 0x7d, 0x72, 0x53, 0xb3, 0x3c, 0xd1, 0x1b, 0x7d, 0xd8, 0xa9, 0xa0, 0x4e, 0x55, 0xf5, 0x72, 0x43, 0x94, 0xaf, 0x27, 0x6c, 0x5c
+    };
+    auto digest = Crypto::Hash::SHA256::hash("0123456789012345678901234567890123456789012345678901234567890123abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcd"sv);
+    EXPECT(memcmp(result, digest.data, Crypto::Hash::SHA256::digest_size()) == 0);
```

Comment:
Only allowing raw `memcmp` here because these tests already use it, we have `Span` and such for a nicer C++ API.